### PR TITLE
Improve reliability of GatewayLookup* tests

### PR DESCRIPTION
**Pull request by akarpz**
### What does this PR do?

Uses `require.Eventually` to retry DNS lookups. Refactors a bit to remove redundant DNS helper methods.

### Motivation

https://datadoghq.atlassian.net/browse/NPM-3012
https://datadoghq.atlassian.net/browse/NPM-3073

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/testutil/testdns/test_dns_server.go [hunk 1]
```diff
@@ -125,3 +127,69 @@ func respond(req *dns.Msg, writer dns.ResponseWriter, record string) {
...
+
+	var dnsClientAddr net.Addr
+	if protocol == "tcp" {
+		dnsClientAddr = &net.TCPAddr{IP: net.ParseIP(clientIP)}
+	} else {
+		dnsClientAddr = &net.UDPAddr{IP: net.ParseIP(clientIP)}
+	}
+
+	localAddrDialer := &net.Dialer{
+		LocalAddr: dnsClientAddr,
```
[the remark this answers is not in the document]
Comment (akarpz) [reply pkg/network/tracer/testutil/testdns/test_dns_server.go:143]: this was moved over from another file.. we probably don't I can test removing it locally to simplify this a bit

[the remark this answers is not in the document]
Comment (hmahmood) [reply pkg/network/tracer/testutil/testdns/test_dns_server.go]: The timeout is not needed?

[the remark this answers is not in the document]
Comment (akarpz) [reply pkg/network/tracer/testutil/testdns/test_dns_server.go]: I moved it to the creation of the client on the line below

[the remark this answers is not in the document]
Comment (akarpz) [reply pkg/network/tracer/testutil/testdns/test_dns_server.go]: I also reduced it from 5 seconds to 3 seconds, and set the `require.Eventually` timeout to 6 seconds. So we should always at least get two tries

**Review by hmahmood — APPROVED** [review]
lgtm pending ci pass

File: pkg/network/tracer/tracer_linux_test.go [hunk 2]
```diff
@@ -659,7 +644,15 @@ func (s *TracerSuite) TestGatewayLookupEnabled() {
 
 	initTracerState(t, tr)
 
-	dnsClientAddr, dnsServerAddr := doDNSQuery(t, "google.com", "8.8.8.8")
+	var clientIP string
+	var clientPort int
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		clientIP, clientPort, _, err = testdns.SendDNSQueries(t, []string{"google.com"}, dnsAddr, "udp")
+		assert.NoError(c, err)
+	}, 6*time.Second, 100*time.Millisecond, "failed to find connection")
```
Comment (hmahmood) [inline pkg/network/tracer/tracer_linux_test.go:652]: "failed to send dns query"?

Comment (hmahmood) [reply pkg/network/tracer/tracer_linux_test.go:652]: Same for a couple of other instances below.

**Review by hmahmood — APPROVED** [review]

**Pull request merged**